utils/weather.py

```diff
@@ -75,8 +75,13 @@ def find_geolocation(self) -> None:
         lat: str = res_location.get("lat")
         long: str = res_location.get("lon")
 
+        # Sets the location attributes, if region is not found, it will use the country
+        # name. If no country name is found, it will then throw an exception.
         self.location: str = res_location.get("name")
-        self.region: str = res_location.get("region")
+        self.region: str = res_location.get("region") or res_location.get("country")
+        if not self.region:
+            log.error("geolocation: %s", "No region or country found")
```

Comment:
I'm not sure logging an error si appropriate here.   This isn't an error in the code, it's handling edge cases from upstream.  I'll ask this... would you need this information when debugging an issue or is the error message returned to the user sufficient enough?

Reply:
No, you’re right, it is not needed here! i can remove that.